core/src/main/java/tachyon/worker/hierarchy/StorageDir.java

```diff
@@ -301,6 +304,9 @@ public boolean deleteBlock(long blockId) throws IOException {
       }
       deleteBlockId(blockId);
     } else {
+      if (mRemovedBlockIdList.contains(blockId)) {
+        mRemovedBlockIdList.remove(blockId);
+      }
```

Comment:
Yeah, because it should only get reported when the block is unlocked and then actually removed.